ast-nodes/expression.dart

```diff
@@ -9,8 +9,8 @@ import '../syntax-error.dart';
 abstract class Expression implements Statement {
   factory Expression.parse(Iterable<Token> tokens) {
     var iterator = tokens.iterator;
-
-    if (!iterator.moveNext()) {
```

Comment:
This is redundant, the two versions have identical null-safety behaviour, but the previous one was more concise.